TransformationSystem/Agent/DataRecoveryAgent.py

```diff
@@ -0,0 +1,538 @@
+"""An agent to ensure consistency for transformation jobs, tasks and files.
+
+Depending on what is the status of a job and its input and output files different actions are performed.
+
+.. warning:: Before fully enabling this agent make sure that your transformation jobs fulfill the assumptions of the
+  agent.  Otherwise it might delete some of your data! Do not set ``Enable`` to True before letting the agent run
+  through a few times and read the messages it produces.
+
+.. note::  Use the :ref:`dirac-transformation-recover-data` script for checking individual transformations
+
+The agent takes the following steps
+
+- obtain list of transformation
+- get a list of all 'Failed' and 'Done' jobs, jobs with pending requests are ignored.
+- get input files for all jobs, get the transformation file status
+  associated for the file (Unused, Assigned, MaxReset, Processed),
+  check if the input file exists
+- get the output files for each job, check if the output files exist
+- perform changes for Jobs, Files and Tasks: cleanup incomplete output
+  files to obtain consistent state for jobs, tasks, input and output
+  files
+
+- Send email about performed actions
+
+Requirements/Assumptions:
+
+- JobParameters:
+
+  - ProductionOutputData: with the list semi-colon separated list of expected output files, stored as a Job Parameter
```

Comment:
This is the last use of "Production" in the agent and related files. Should I add an option to change the parameter under which the OutputData files can be found?

Reply:
I am afraid that this is used in so many places that changing it would risky...